Missing import

### Describe the bug

I get the following error when I try the tutorial [here](https://craby.rs/docs/get-started/create-project)


error[E0432]: unresolved import `crate::calculator_impl`
 --> crates/lib/src/ffi.rs:5:12
  |
5 | use crate::calculator_impl::*;
  |            ^^^^^^^^^^^^^^^ could not find `calculator_impl` in the crate root

error[E0432]: unresolved import `super`
  --> crates/lib/src/ffi.rs:13:14
   |
13 |         type Calculator;
   |              ^^^^^^^^^^ no `Calculator` in `ffi`

warning: unused import: `bridging::*`
 --> crates/lib/src/ffi.rs:8:5
  |
8 | use bridging::*;
  |     ^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `crate::ffi::bridging::*`
 --> crates/lib/src/generated.rs:6:5
  |
6 | use crate::ffi::bridging::*;
  |     ^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0432`.
warning: `calculator` (lib) generated 2 warnings
error: could not compile `calculator` (lib) due to 2 previous errors; 2 warnings emitted

ERROR Failed to build (Target: aarch64-linux-android)

The solution for me is to manually add pub(crate) mod calculator_impl; to crates/lib/src/lib.rs, but I assume it's meant to be added automatically.

### Reproduction (Optional)

_No response_

### Steps to reproduce

Follow the steps in [this tutorial ](https://craby.rs/docs/get-started/create-project). You'll get an 'unresolved import' error when running step 4.

### Rust Version

```shell

```

### Platform Information

_No response_

### Doctor Output

_No response_

### macOS Version

_No response_

### Package Versions

_No response_

### Logs

_No response_

--- a/crates/craby_cli/src/commands/codegen/handler.rs
+++ b/crates/craby_cli/src/commands/codegen/handler.rs
@@ -144,13 +144,11 @@ pub fn perform(opts: CodegenOptions) -> anyhow::Result<()> {
     Ok(())
 }
 
-/// Checks `lib.rs` for missing or stale `mod <impl>` declarations.
+/// Checks `lib.rs` for missing `mod <impl>` declarations.
 ///
-/// Since `lib.rs` is never overwritten (to preserve user code), module additions
-/// or removals won't be reflected automatically. This function warns when:
-///
-/// - An expected `mod <name>_impl;` is absent from the file (module added to spec)
-/// - A `mod <name>_impl;` exists in the file but is no longer in the schema (module removed)
+/// Since `lib.rs` is never overwritten (to preserve user code), adding a new
+/// craby module won't be reflected automatically. This function warns when an
+/// expected `mod <name>_impl;` is absent from the file.
 fn check_lib_rs_mods(ctx: &CodegenContext) {
     let lib_rs_path = crate_dir(&ctx.root).join("src").join("lib.rs");
 
@@ -159,44 +157,12 @@ fn check_lib_rs_mods(ctx: &CodegenContext) {
         Err(_) => return,
     };
 
-    let expected: Vec<String> = ctx
-        .schemas
-        .iter()
-        .map(|s| impl_mod_name(&s.module_name))
-        .collect();
-
-    // Warn about missing declarations
-    for mod_name in &expected {
+    for mod_name in ctx.schemas.iter().map(|s| impl_mod_name(&s.module_name)) {
         if !content.contains(&format!("mod {mod_name}")) {
             warn!(
                 "lib.rs is missing module declaration: `pub(crate) mod {mod_name};`\n  → Add it manually or delete lib.rs to regenerate."
             );
         }
-    }
-
-    // Warn about stale declarations (ends with `_impl` but no longer in schema)
-    for line in content.lines() {
-        if let Some(mod_name) = parse_mod_name(line.trim()) {
-            if mod_name.ends_with("_impl") && !expected.iter().any(|m| m == mod_name) {
-                warn!(
-                    "lib.rs contains a stale module declaration: `mod {mod_name};`\n  → Remove it manually."
-                );
-            }
-        }
-    }
-}
-
-/// Extracts the module name from a `mod` declaration line.
-///
-/// Matches: `mod foo;`, `pub mod foo;`, `pub(crate) mod foo;`
-fn parse_mod_name(line: &str) -> Option<&str> {
-    let line = line.strip_suffix(';')?.trim();
-    let idx = line.rfind("mod ")? + 4;
-    let name = line[idx..].trim();
-    if !name.is_empty() && name.chars().all(|c| c.is_alphanumeric() || c == '_') {
-        Some(name)
-    } else {
-        None
     }
 }
 

--- a/crates/craby_cli/src/commands/codegen/handler.rs
+++ b/crates/craby_cli/src/commands/codegen/handler.rs
@@ -15,8 +15,8 @@ use craby_codegen::{
     },
     types::CodegenContext,
 };
-use craby_common::{config::load_config, constants::craby_tmp_dir, env::is_initialized};
-use log::{debug, info};
+use craby_common::{config::load_config, constants::{crate_dir, craby_tmp_dir, impl_mod_name}, env::is_initialized};
+use log::{debug, info, warn};
 use owo_colors::OwoColorize;
 
 use crate::utils::{file::write_file, schema::print_schema};
@@ -134,12 +134,70 @@ pub fn perform(opts: CodegenOptions) -> anyhow::Result<()> {
         }
     }
 
+    check_lib_rs_mods(&ctx);
+
     info!(
         "Codegen completed successfully 🎉 {}",
         format!("({}ms)", elapsed).dimmed()
     );
 
     Ok(())
+}
+
+/// Checks `lib.rs` for missing or stale `mod <impl>` declarations.
+///
+/// Since `lib.rs` is never overwritten (to preserve user code), module additions
+/// or removals won't be reflected automatically. This function warns when:
+///
+/// - An expected `mod <name>_impl;` is absent from the file (module added to spec)
+/// - A `mod <name>_impl;` exists in the file but is no longer in the schema (module removed)
+fn check_lib_rs_mods(ctx: &CodegenContext) {
+    let lib_rs_path = crate_dir(&ctx.root).join("src").join("lib.rs");
+
+    let content = match std::fs::read_to_string(&lib_rs_path) {
+        Ok(c) => c,
+        Err(_) => return,
+    };
+
+    let expected: Vec<String> = ctx
+        .schemas
+        .iter()
+        .map(|s| impl_mod_name(&s.module_name))
+        .collect();
+
+    // Warn about missing declarations
+    for mod_name in &expected {
+        if !content.contains(&format!("mod {mod_name}")) {
+            warn!(
+                "lib.rs is missing module declaration: `pub(crate) mod {mod_name};`\n  → Add it manually or delete lib.rs to regenerate."
+            );
+        }
+    }
+
+    // Warn about stale declarations (ends with `_impl` but no longer in schema)
+    for line in content.lines() {
+        if let Some(mod_name) = parse_mod_name(line.trim()) {
+            if mod_name.ends_with("_impl") && !expected.iter().any(|m| m == mod_name) {
+                warn!(
+                    "lib.rs contains a stale module declaration: `mod {mod_name};`\n  → Remove it manually."
+                );
+            }
+        }
+    }
+}
+
+/// Extracts the module name from a `mod` declaration line.
+///
+/// Matches: `mod foo;`, `pub mod foo;`, `pub(crate) mod foo;`
+fn parse_mod_name(line: &str) -> Option<&str> {
+    let line = line.strip_suffix(';')?.trim();
+    let idx = line.rfind("mod ")? + 4;
+    let name = line[idx..].trim();
+    if !name.is_empty() && name.chars().all(|c| c.is_alphanumeric() || c == '_') {
+        Some(name)
+    } else {
+        None
+    }
 }
 
 fn with_generated_comment(path: &Path, code: &str) -> String {

--- a/examples/craby-test/cpp/CxxCrabyTestModule.cpp
+++ b/examples/craby-test/cpp/CxxCrabyTestModule.cpp
@@ -453,11 +453,11 @@ jsi::Value CxxCrabyTestModule::promiseStringMethod(jsi::Runtime &rt,
     }
 
     auto arg0$raw = args[0].asString(rt).utf8(rt);
-    auto arg0 = rust::Str(arg0$raw.data(), arg0$raw.size());
     react::AsyncPromise<double> promise(rt, callInvoker);
 
-    thisModule.threadPool_->enqueue([it_, promise, arg0]() mutable {
+    thisModule.threadPool_->enqueue([it_, promise, arg0$raw]() mutable {
       try {
+        auto arg0 = rust::Str(arg0$raw.data(), arg0$raw.size());
         auto ret = craby::crabytest::bridging::promiseStringMethod(*it_, arg0);
         promise.resolve(ret);
       } catch (const jsi::JSError &err) {

